### PR TITLE
Complete set up script to build staging and production environment se…

### PR DESCRIPTION
**Pull request by noahsilencelab**
# Why?
- We have 2 `staging`, and `production` Snaps as 2 different NPM packages.
- We need to change the content of `package.json`, `snap.manifest.json` before bundling and publishing.
# How?
- Create `scripts/pre-publish.js` to change the content of `package.json`, `snap.manifest.json` dynamically based on `NODE_ENV` value
- Create 2 `snap.config.js` and `snap.staging.config.js` files to configure Snap for 2 environments. The config files allow Snap to load different sets of environment variables while being bundled.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,87 +1,93 @@
...
+    "images/",
+    "snap.manifest.json"
+  ],
+  "scripts": {
+    "build": "yarn build:prod",
+    "build:prod": "mm-snap build -c ./snap.config.ts",
+    "build:stg": "mm-snap build -c ./snap.staging.config.ts",
+    "build:clean:prod": "yarn clean && yarn build:prod",
+    "build:clean:stg": "yarn clean && yarn build:stg",
+    "prepublish:prod": "NODE_ENV='production' node ./scripts/pre-publish.js && yarn build:clean:prod",
```
Comment (dakshsilence) [inline package.json]: need to run `yarn install` again because package name is changed.

Comment (dakshsilence) [reply package.json]:
```suggestion
    "prepublish:prod": "NODE_ENV='production' node ./scripts/pre-publish.js && yarn install && yarn build:clean:prod",
```

File: scripts/pre-publish.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const execEnv = process.env.NODE_ENV;
+const currentVersion = packageJson.version;
+let baseRepoUrl = '';
+let name = '';
+let rpcOrigins = [];
+let keyringOrigins = [];
+
+// Check the environment
+if (execEnv === 'staging') {
+	console.log(`${chalk.green.bold('PREPUBLISH STAGING ANNOUNCEMENT')}\u{1F447}\u{1F447}\u{1F447}`);
```
Comment (leontiad) [inline scripts/pre-publish.js:20]: what is that hardcoded byted?\u{1F447}\u{1F447}\u{1F447}

File: scripts/pre-publish.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	baseRepoUrl =
+		'https://www.npmjs.com/package/@silencelaboratories/silent-shard-snap-staging';
+	name = '@silencelaboratories/silent-shard-snap-staging';
+    rpcOrigins = ["https://snap-staging.silencelaboratories.com", "http://localhost:3000"];
+    keyringOrigins = ["https://snap-staging.silencelaboratories.com", "http://localhost:3000"];
+} else if (execEnv === 'production') {
+	console.log(`${chalk.green.bold('PREPUBLISH PRODUCTION ANNOUNCEMENT')}\u{1F447}\u{1F447}\u{1F447}`);
```
Comment (leontiad) [inline scripts/pre-publish.js:27]: same

File: scripts/pre-publish.js (hunk 2 above)
Comment (noahsilencelab) [reply scripts/pre-publish.js:20]: https://www.compart.com/en/unicode/U+1F447 That's the icon to make the stdout look nice

File: scripts/pre-publish.js (hunk 3 above)
Comment (noahsilencelab) [reply scripts/pre-publish.js:27]: https://www.compart.com/en/unicode/U+1F447 That's the icon to make the stdout look nice

**Review by leontiad — APPROVED** [review]
LGTM

**Pull request merged**